### PR TITLE
Check the length of regex in a dynamic path

### PR DESCRIPTION
**Pull request by byzheng**
solves #347 and #348

### CODE REVIEW

**Comment by schloerke** [comment]
@byzheng The PR is currently "failing" as code coverage is decreased. (even if < 1%).  This is not a stopper for me.

I really like the PR.  Could you add a couple tests to validate that type `chr` works?

Thank you!

**Comment by byzheng** [comment]
@schloerke  Add one test for chr. Not sure whether it is enough for you.

File: R/query-string.R [hunk 1]
```diff
@@ -71,10 +71,11 @@ createPathRegex <- function(pathDef){
 
 typeToRegex <- function(type){
```
Comment (wch) [inline R/query-string.R]: A little simpler and clearer to do something like:

```R
if (length(type) == 0)
  return(character(0))
# Do the rest of the logic here
```

File: R/swagger.R [hunk 2]
```diff
@@ -1,6 +1,9 @@
 #' Parse the given plumber type and return the typecast value
 #' @noRd
 plumberToSwaggerType <- function(type){
+  if (is.na(type)) {
```
Comment (wch) [inline R/swagger.R]: Why is this necessary now?

File: R/query-string.R [hunk 3]
```diff
@@ -71,10 +71,11 @@ createPathRegex <- function(pathDef){
...
   re <- rep("[^/]+", length(type))
```
Comment (wch) [inline R/query-string.R]: It would help to have a comment explaining which cases get this default value (like `NA` and `"chr"`?)
